core/common/src/Instant.kt

```diff
@@ -96,29 +268,39 @@ public expect class Instant : Comparable<Instant> {
      *
      * The result is never clamped, but note that for instants that are far apart,
      * the value returned may represent the duration between them inexactly due to the loss of precision.
+     *
+     * Note that sources of [Instant] values (in particular, [Clock]) are not guaranteed to be in sync with each other
+     * or even monotonic, so the result of this operation may be negative even if the other instant was observed later
+     * than this one, or vice versa.
```

Comment:
Putting myself in the mindset of someone who doesn't know how to measure time intervals in Kotlin but does know about `Instant` values, I can see myself obtaining two values first and then reading the documentation of subtraction to see if it will give me the elapsed time. Do you see a better way to warn people in these situations?

Reply:
I think it'd be helpful for people unfamiliar with the quirks of time and system clocks to have a practical example here. Eg: 

> For example, systems using [Network Time Protocol](https://en.wikipedia.org/wiki/Network_Time_Protocol) may experience discontinuous jumps in time both forwards and backwards after synchronizing with a reference clock. In those scenarios, the result of this operation could be greater than or less than what's expected. 

I like practical examples like this because it allows people reading the documentation to read about other systems which might influence kotlinx-datetime. While the discontinuous jumps aren't limited to NTP syncs, it does speak to this in a less abstract and difficult to parse manner.